src/pe_reports/helpers/bulletin/bulletin_generator.py

```diff
@@ -0,0 +1,274 @@
+
+    cur.callproc("query_emails", [breach, org_name])
+    emails = cur.fetchall()
+    cols = [desc[0] for desc in cur.description]
+    cur.close()
+    conn.close()
+    emails_df = pd.DataFrame(emails, columns=cols)
+
+    emails_df = emails_df.rename(
+        columns={
+            "email": "Email",
+            "name": "Name",
+            "login_id": "Login ID",
+            "phone": "Phone",
+            "password": "Password",
+            "hash_type": "Hash Type",
+        }
+    )
+
+    # Replace Blank values with DataFrame.replace() methods.
+    emails_df = emails_df.replace(r"^\s*$", "-", regex=True)
+    df_table = emails_df[
+        ["Email", "Name", "Login ID", "Phone", "Password", "Hash Type"]
+    ].to_html(index=False, classes="table table-striped")
+    emails_list = emails_df["Email"].values.tolist()
+
+    emails_list.sort()
+    email_count = len(emails_list)
+    n = 2
+    hibp_rows = [
+        emails_list[i * n : (i + 1) * n] for i in range((len(emails_list) + n - 1) // n)
+    ]
+
+    results = []
+    for row in breaches:
+        row_dict = {}
+        for i, col in enumerate(column_names):
+            row_dict[col] = row[i]
+        results.append(row_dict)
+
+        outputText = template.render(
+            row_dict,
+            user_provided_content=user_text,
+            email_count=email_count,
+            df_table=df_table,
+            hibp_rows=hibp_rows,
+        )
+        html_file = open(bulletin_path + "/creds_bulletin_template_filled.html", "w")
+        html_file.write(outputText)
+        html_file.close()
+
+        options = {
+            "page-size": "Letter",
+            "margin-top": "1.3in",
+            "margin-right": "0in",
+            "margin-bottom": "1.2in",
+            "margin-left": "0in",
+            "encoding": "UTF-8",
+            "custom-header": [("Accept-Encoding", "gzip")],
+            "cookie": [
+                ("cookie-empty-value", '""'),
+                ("cookie-name1", "cookie-value1"),
+                ("cookie-name2", "cookie-value2"),
+            ],
+            "no-outline": None,
+            "header-html": bulletin_path + "/header.html",
+            "header-spacing": -5,
+            "footer-right": "[page] of [topage]       \t\t\t\t",
+            "footer-left": "       \t\t\tPosture & Exposure",
+            "footer-spacing": 8,
+            "footer-html": bulletin_path + "/footer.html",
+            "enable-local-file-access": True,
+            "disable-smart-shrinking": True,
+        }
+        out_path = output_directory + "/" + filename
+
+        pdfkit.from_file(
+            [bulletin_path + "/creds_bulletin_template_filled.html"],
+            out_path,
+            options=options,
+            verbose=True,
+        )
+
+
+def main():
+    """Generate a bulletin for a cybersixgill post."""
+    args = docopt(__doc__)
+
+    id = args["INTEL_ITEM_ID"]
+
+    user_text = html_builder("")
+
+    LOGGER.info(f"Running on {id}")
```

Comment:
We avoid putting f-strings in our logging statements (explanation [here](https://blog.devgenius.io/when-python-old-style-string-formatting-is-best-practice-691a0759b37e)):
```suggestion
    LOGGER.info("Running on %s", id)
```